pkg/infinispan/cluster.go

```diff
@@ -88,7 +88,7 @@ func (c Cluster) GetClusterSize(podName string) (int, error) {
 // GracefulShutdown performs clean cluster shutdown
 func (c Cluster) GracefulShutdown(podName string) error {
 	rsp, err, reason := c.Client.Post(podName, consts.ServerHTTPClusterStop, "", nil)
-	return validateResponse(rsp, reason, err, "during graceful shutdown", http.StatusOK)
+	return validateResponse(rsp, reason, err, "during graceful shutdown", http.StatusNoContent)
```

Comment:
maybe it's for a differen PR but...
how about a more relaxed check status in 200..299? 
I saying this because I've already seen this kind of changes and return code isn't strictly specified server side.

Reply:
Sorry, @rigazilla , but I don't agree with you in this case. Return codes should be strictly specified for implementation as well as documentation also.
That's the reason why I've reviewed whole REST API in PR https://github.com/infinispan/infinispan/pull/9040 for correct return codes to be specified. If API will change we will never know about it with this kind of relaxations.

Reply:
well in my opinion a generic success return code would be sufficient in most of the cases.
Anyway it's also ok with me if a well define interface is established server side.

Reply:
Yes, agreed about most cases, but in some cases we need detailed code validation, not in range